utilities/cli/holohub.py

```
@@ -1632,231 +1631,183 @@ def handle_autocompletion_list(self, args: argparse.Namespace) -> None:
             )
-            holohub_cli_util.run_command(
+            sys.exit(1)
+
+        return target
+
+    def _collect_lint_files(self, target: Path) -> List[str]:
+        """Collect git-tracked and unignored files for ``pre-commit run --files``."""
+        root = HoloHubCLI.HOLOHUB_ROOT.resolve()
+        target_arg = "." if target == root else str(target.relative_to(root))
+        try:
+            output = subprocess.check_output(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)